### ISSUE TITLE
rucio client container missing things

### ISSUE THREAD

**Issue by ivukotic** [rucio/containers#71]
I tried to use the latest docker image and it does not work:

rucio whoami
Traceback (most recent call last):
  File "/usr/bin/rucio", line 64, in <module>
    from rucio.client.uploadclient import UploadClient
  File "/usr/lib/python2.7/site-packages/rucio/client/uploadclient.py", line 41, in <module>
    from rucio.rse import rsemanager as rsemgr
  File "/usr/lib/python2.7/site-packages/rucio/rse/__init__.py", line 94, in <module>
    from rucio.core.rse import get_rse_protocols, get_rse_id
ImportError: No module named core.rse

**Comment by tbeerman** [comment]
Hi Ilija,
can you please give a bit more details? How do you run the container? I cannot really reproduce this, for me the container works, also the autogenerated rucio.cfg is working despite the wrong path.

**Comment by ivukotic** [comment]
Hi,
I am using your image to base my image on it. That will be a frequent use case as more and more applications move to k8s. I am not sure how is one supposed to a set everything up. 
First I had an issue with the wrong path, since my image sets WORKDIR /usr/src/app.
In this case bashrc script to generate rucio.cfg won't work. So the first fix is to make it absolute path.
The second problem I had was to figure out what exactly I need to authorize. 
Can I do it all using ENV variables or rucio.cfg is always needed?
Errors given were never correct. 
It took me way too much time to set it up. Finally I took a ATLAS config template from CVMFS. IMHO default configs for larger experiments should be delivered with the client.

**Comment by pchanial** [comment]
I have most probably encountered the same problem. Except that now the init scripts that create the file rucio.cfg are in /etc/profile.d

With this Dockerfile:
```Dockerfile
FROM rucio/rucio-clients:release-1.23.10_py3
COPY run_scripts.sh .
ENTRYPOINT ["./run_scripts.sh"]
```
and run_scripts.sh:
```bash
#!/bin/bash
echo "Running scripts..."
```
If I run...
```bash
docker run --entrypoint bash -it --rm pchanial/rucio-issue71
```
...the init scripts are run but not `run_scripts.sh`. While running
```bash
docker run -it --rm pchanial/rucio-issue71
```
...the init scripts are not, but `run_scripts.sh` is.
You can test for yourself, I have uploaded the image pchanial/rucio-issue71 to docker hub.
So basically overriding the ENTRYPOINT has the unexpected consequence of preventing the creation of rucio.cfg.
Why isn't the default rucio.cfg file copied in /opt/rucio/etc in the Dockerfile?

**Comment by tbeerman** [comment]
Hi @pchanial ,

sorry, I completely missed your comment here.
The rucio.cfg has to be generated dynamically and we cannot just copy a default one to /opt/rucio/etc. Things like your Rucio host and the authentication details depend on your own Rucio server setup if we would put some static default rucio.cfg if would just not work. So we generate it based on env variables passed to the container and a jinja2 template. If you want to use you on script as entrypoint you have to make sure to run `/etc/profile.d/rucio_init.sh` once to generate the config.

Hope that helps.
